src/Sylius/Component/Payment/Model/Payment.php

```diff
@@ -69,12 +69,13 @@ public function setCurrencyCode(string $currencyCode): void
 
     public function getAmount(): ?int
     {
-        return $this->amount;
+        return (int) $this->amount;
     }
 
-    public function setAmount(int $amount): void
+    /** @param string|int $amount (bigint in db) */
+    public function setAmount($amount): void
     {
-        $this->amount = $amount;
+        $this->amount = (int) $amount > PHP_INT_MAX ? PHP_INT_MAX : (int) $amount;
```

Comment:
@timching Not a good idea, it is better to just let PHP throw exception than change the logic silently.